src/electron/main.ts

```diff
@@ -22,13 +37,69 @@ function createWindow(): void {
 
 	// and load the index.html of the app.
 	win.loadURL(
-		url.format({
-			pathname: PathUtils.join(__dirname, 'renderer', 'app.html'),
+		Url.format({
+			pathname: Path.join(__dirname, 'app.html'),
 			protocol: 'file:',
 			slashes: true
 		})
 	);
 
+	// tslint:disable-next-line:await-promise
+	const port = await getPort({ port: 1879 });
+	const server = await createServer({ port });
+
+	// tslint:disable-next-line:no-any
+	const send = (message: any) => {
+		if (win) {
+			win.webContents.send('message', message);
+		}
+	};
+
+	// tslint:disable-next-line:no-any
+	ipcMain.on('message', (e: Electron.Event, payload: any) => {
+		if (!payload) {
+			return;
+		}
+
+		server.emit('message', payload);
+
+		switch (payload.type) {
+			case 'app-loaded': {
+				if (win) {
+					win.webContents.send('message', {
+						type: 'start-app',
+						payload: port
+					});
+				}
+			}
+		}
+	});
+
+	// tslint:disable-next-line:no-any
+	server.on('client-message', (envelope: any) => {
```

Comment:
as we assume to get a string here we could also type it as such. Maybe add a comment that a different client message could be triggered and that we need the try/catch because of this